torch/csrc/jit/script/init.cpp

```diff
@@ -312,13 +311,28 @@ struct ModuleValue : public SugaredValue {
       return std::make_shared<ModuleValue>(v->module);
     } else if (Method* v = module->find_method(field)) {
       return std::make_shared<MethodValue>(shared_from_this(), *v);
-    } else if (NamedParameter* v = module->find_parameter(field)) {
+    } else if (NamedInput* v = module->find_parameter(field)) {
       return std::make_shared<SimpleValue>(m.get_or_add_parameter(v->slot()));
+    } else if (NamedInput* v = module->find_attribute(field)) {
+      return std::make_shared<SimpleValue>(
+          m.get_or_add_attribute(v->type, v->slot()));
```

Comment:
`get_or_add_attribute` should take a `NamedInput*` rather than both a type and a IValue*. Otherwise it is too easy to accidentally pass the wrong type.